### DISCUSSION TITLE
JMX and OpenGrok

### DISCUSSION THREAD

**Discussion by ChristopheBordieu** [oracle/opengrok#4115] [Q&A] [answered]
Hi team,

Is there anything special to know before starting OpenGrok 1.7(.35) webapp/indexer JVMs using JMX? Or it is to be done the classical way?

**Comment by vladak** [comment, marked as answer]
Anything JMX related was removed via PR #1343. What is that you want to do exactly ?

**Reply by ChristopheBordieu** [reply]
I want to be able to monitor the webapp JVM. For instance, in order to be alerted (via JMX exporter data) when heap size starts being too small. Typically, since some days, our webapp is facing some troubles. Very likely, we have to increase heap size. But before doing that, we would like to monitor the JVM and look at it to understand whether some new repos that are indexed could be the root cause of these troubles.

**Reply by vladak** [reply]
There are [monitoring](https://github.com/oracle/opengrok/wiki/Monitoring) capabilities built into both webapp and indexer for that. Easy to do with e.g. Grafana alerting or such.

**Reply by ChristopheBordieu** [reply]
Ok.
For the webapp, I have this resource _/metrics/prometheus_ already available and managed by our proxy. I am going to add this exporter to our Grafana instance.
For the indexer, I will dig into this.
Thanks!
